test/jest_integration/jwt_multiauth.test.ts

```diff
@@ -0,0 +1,211 @@
+      order: 0,
+      http_authenticator: {
+        challenge: false,
+        type: 'jwt',
+        config: {
+          signing_key: 'OTkwMTFkZjZlZjQwZTRhMmNkOWNkNmNjYjJkNjQ5ZTAK',
+          jwt_header: 'Authorization',
+          jwt_url_parameter: 'token',
+          jwt_clock_skew_tolerance_seconds: 30,
+          roles_key: 'roles',
+          subject_key: 'sub',
+        },
+      },
+      authentication_backend: {
+        type: 'noop',
+        config: {},
+      },
+    };
+    try {
+      config.dynamic!.authc!.jwt_auth_domain = jwtConfig;
+      config.dynamic!.authc!.basic_internal_auth_domain.http_authenticator.challenge = true;
+      config.dynamic!.http!.anonymous_auth_enabled = false;
+      await wreck.put('https://localhost:9200/_plugins/_security/api/securityconfig/config', {
+        payload: config,
+        rejectUnauthorized: false,
+        headers: {
+          'Content-Type': 'application/json',
+          authorization: ADMIN_CREDENTIALS,
+        },
+      });
+    } catch (error) {
+      console.log('Got an error while updating security config!', error.stack);
+      fail(error);
+    }
+  });
+
+  afterAll(async () => {
+    console.log('Remove the Sample Data');
+    await wreck
+      .delete('http://localhost:5601/api/sample_data/flights', {
+        rejectUnauthorized: false,
+        headers: {
+          'Content-Type': 'application/json',
+          authorization: ADMIN_CREDENTIALS,
+        },
+      })
+      .then((value) => {
+        Promise.resolve(value);
+      })
+      .catch((value) => {
+        Promise.resolve(value);
+      });
+    console.log('Remove the Security Config');
+    await wreck
+      .patch('https://localhost:9200/_plugins/_security/api/securityconfig', {
+        payload: [
+          {
+            op: 'remove',
+            path: '/config/dynamic/authc/jwt_auth_domain',
+          },
+        ],
+        rejectUnauthorized: false,
+        headers: {
+          'Content-Type': 'application/json',
+          authorization: ADMIN_CREDENTIALS,
+        },
+      })
+      .then((value) => {
+        Promise.resolve(value);
+      })
+      .catch((value) => {
+        Promise.resolve(value);
+      });
+    // shutdown OpenSearchDashboards server
+    await root.shutdown();
+  });
+
+  it('Verify JWT access to dashboards', async () => {
+    console.log('Wreck access home page');
+    const adminJWT = await new SignJWT({
+      roles: [JWT_ADMIN_ROLE],
+      sub: ADMIN_USER,
+    })
+      .setProtectedHeader({ alg: 'HS256' })
+      .sign(new TextEncoder().encode('99011df6ef40e4a2cd9cd6ccb2d649e0'));
+    await wreck
+      .get(`http://localhost:5601/app/home?token=${adminJWT}#`, {
+        rejectUnauthorized: true,
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      })
+      .then((value) => {
+        Promise.resolve(value);
+      })
+      .catch((value) => {
+        Promise.resolve(value);
+      });
+  });
```

Comment:
Would that be a test using cypress perhaps?

Reply:
I think we can probably check the status code of the get request to `http://localhost:5601/app/login`, but I am ok with pushing this off into a follow up. Also do we want to change the catch block so it would fail the test? I think then and catch should have different behavior here

Reply:
Working on it.

Reply:
I'm trying to run the Jest tests with `yarn test:jest_server`, but it's failing with 403 errors when trying to update the security config in OpenSearch. I thought the security config has to be updated by a super-admin, how is this supposed to work?

Reply:
@cwperks ?

Reply:
It works in the integration tests because of this setting: https://github.com/derek-ho/start-opensearch/blob/main/action.yml#L127

`plugins.security.unsupported.restapi.allow_securityconfig_modification: true`

That setting allows the test to update the securityconfig. See an [example](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/test/jest_integration/jwt_auth.test.ts#L122-L153) from jwt_auth.test.ts.

Reply:
Thanks, that helped with the 403 errors. Just pushed a commit with the above updates.